pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileProcessor.java

```diff
@@ -73,26 +73,28 @@ public static LogFileSequenceProcessor create(
         final TransactionCoordinator transactionCoordinator = new TransactionCoordinator(connection, writer);
         transactionCoordinator.performRecovery();
 
-        final EventGenerator eventGenerator = EventGenerator.create(
-                config.routingKey,
-                config.maxRecordsPerEvent,
-                config.eventTemplateStr,
-                writerId);
         final TransactionStateSQLiteImpl state = new TransactionStateSQLiteImpl(connection, transactionCoordinator);
-        return new LogFileSequenceProcessor(config, state, writer, transactionCoordinator, eventGenerator);
+        return FileProcessorFactory.createFileSequenceProcessor(config, state, writer, transactionCoordinator,writerId);
+
     }
 
-    public void watchLogFiles() throws Exception {
+    /* The abstract method serves as an Event Generator responsible for generating events.
+     * This logic is tailored to specific file types, and as such, it will be implemented in their respective classes.
+     * @param config configurations parameters
+     * @return eventGenerator
+     */
+    public abstract EventGenerator getEventGenerator(FileConfig config);
+    public void watchFiles() throws Exception {
         findAndRecordNewFiles();
     }
-    public void processLogFiles() throws Exception {
-        log.info("processLogFiles: BEGIN");
-        processNewFiles();
+    public void processFiles() throws Exception {
+        log.info("processFiles: BEGIN");
         if (config.enableDeleteCompletedFiles) {
-            log.debug("processLogFiles: Deleting completed files");
+            log.debug("processFiles: Deleting completed files");
             deleteCompletedFiles();
         }
-        log.info("processLogFiles: END");
+        processNewFiles();
+        log.info("processFiles: END");
     }
 
     public void processNewFiles() throws Exception {
```

Comment:
In case of no new files, the logs are filled up with these messages repeatedly
```
INFO  [2023-11-14 13:57:37.737] [FileIngestService-PARQ2-0] i.p.s.collector.file.FileProcessor: processFiles: BEGIN
INFO  [2023-11-14 13:57:37.737] [FileIngestService-PARQ2-0] i.p.s.collector.file.FileProcessor: processNewFiles: No more files to watch    
INFO  [2023-11-14 13:57:37.737] [FileIngestService-PARQ2-0] i.p.s.collector.file.FileProcessor: processFiles: END
INFO  [2023-11-14 13:57:37.753] [FileIngestService-PARQ2-0] i.p.s.collector.file.FileProcessor: processFiles: BEGIN
INFO  [2023-11-14 13:57:37.753] [FileIngestService-PARQ2-0] i.p.s.collector.file.FileProcessor: processNewFiles: No more files to watch    
INFO  [2023-11-14 13:57:37.753] [FileIngestService-PARQ2-0] i.p.s.collector.file.FileProcessor: processFiles: END
```

Reply:
Changed BEGIN/END logging level to debug. Can we retain "No more files to watch" under **Info** level? This way it serves to inform that, currently there are no new files in system awaiting processing. Does that make sense?